cli/flags.rs

```diff
@@ -2041,10 +2046,13 @@ fn no_check_arg<'a>() -> Arg<'a> {
     .min_values(0)
     .value_name("NO_CHECK_TYPE")
     .long("no-check")
-    .help("Skip type checking modules")
+    .help("DEPRECATED: Skip type checking modules")
     .long_help(
-      "Skip type-checking. If the value of '--no-check=remote' is supplied, \
-      diagnostic errors from remote modules will be ignored.",
+      "DEPRECATED: Skip type-checking. If the value of '--no-check=remote' is supplied,
+diagnostic errors from remote modules will be ignored.
+
+This option will be removed in Deno 1.23, see `--check` instead.
+      ",
```

Comment:
this seems to be doing more than just "type-check only local files by default"

Reply:
I don't understand what you mean here